src/HeaderCell.tsx

```diff
@@ -64,6 +65,10 @@ export default function HeaderCell<R, SR>({
     [cellResizableClassname]: column.resizable
   });
 
+  const defaultComponents = useDefaultComponents<R, SR>();
+  const HeaderRenderer =
+    column.headerRenderer ?? defaultComponents?.headerRenderer ?? DefaultHeaderRenderer;
```

Comment:
If we move `?? DefaultHeaderRenderer;` in DataGrid.tsx we'd avoid an extra check per header cell 🤔